diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index 2f3405c..6bf4da5 100644
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -335,7 +335,7 @@ endfunction
 
 function! s:fill_quickfix(list, ...)
   if len(a:list) > 1
-    call setqflist(a:list)
+    call setqflist([], ' ', {'items': sort(a:list, 'tomtomjhj#qf#compare_quickfix_item'), 'title': 'fzf'})
     copen
     wincmd p
     if a:0
