diff --git a/autoload/fzf/vim.vim b/autoload/fzf/vim.vim
index 078c6e8..e34f948 100755
--- a/autoload/fzf/vim.vim
+++ b/autoload/fzf/vim.vim
@@ -209,7 +209,7 @@ function! fzf#vim#with_preview(...)
   end
 
   if len(args)
-    call extend(preview, ['--bind', join(map(args, 'v:val.":toggle-preview"'), ',')])
+    call extend(preview, ['--bind', join(map(args, 'v:val.":toggle-preview"') + ['ctrl-alt-j:preview-half-page-down', 'ctrl-alt-k:preview-half-page-up'] , ',')])
   endif
   call s:merge_opts(spec, preview)
   return spec
