diff --git a/lua/dap-view/breakpoints/view.lua b/lua/dap-view/breakpoints/view.lua
index 7b78b10..9e99582 100644
--- a/lua/dap-view/breakpoints/view.lua
+++ b/lua/dap-view/breakpoints/view.lua
@@ -36,7 +36,6 @@ M.show = function()
                 local col_offset = #relative_path + #tostring(entry.lnum) + 2
 
                 treesitter.copy_highlights(buf, entry.lnum - 1, line, col_offset)
-                extmarks.copy_extmarks(buf, entry.lnum - 1, line, col_offset)
 
                 hl.highlight_file_name_and_line_number(line, #relative_path, #tostring(entry.lnum))
 
