diff --git a/lua/cmp/utils/keymap.lua b/lua/cmp/utils/keymap.lua
index 5ab0be7..797a5bc 100644
--- a/lua/cmp/utils/keymap.lua
+++ b/lua/cmp/utils/keymap.lua
@@ -16,7 +16,9 @@ end
 ---Normalize key sequence.
 ---@param keys string
 ---@return string
-keymap.normalize = function(keys)
+keymap.normalize = vim.fn.has('nvim-0.8') == 1 and function(keys)
+  return vim.fn.keytrans(vim.api.nvim_replace_termcodes(keys, true, true, true))
+end or function(keys)
   local normalize_buf = buffer.ensure('cmp.util.keymap.normalize')
   vim.api.nvim_buf_set_keymap(normalize_buf, 't', keys, '<Plug>(cmp.utils.keymap.normalize)', {})
   for _, map in ipairs(vim.api.nvim_buf_get_keymap(normalize_buf, 't')) do
