diff --git a/lua/dap-view/views/util.lua b/lua/dap-view/views/util.lua
index 3f3baf4..8fe55f1 100644
--- a/lua/dap-view/views/util.lua
+++ b/lua/dap-view/views/util.lua
@@ -54,22 +54,14 @@ M.jump_to_location = function(pattern, column)
     local switchbufopt = config.switchbuf
     local win = window.get_win_respecting_switchbuf(switchbufopt, bufnr)
 
-    if not win then
-        win = api.nvim_open_win(0, true, {
-            split = util.inverted_directions[config.windows.position],
-            win = -1,
-            height = config.windows.height < 1 and math.floor(vim.go.lines * (1 - config.windows.height))
-                or math.floor(vim.go.lines - config.windows.height),
-        })
+    if win then
+      api.nvim_set_current_win(win)
+      api.nvim_set_current_buf(bufnr)
+    else
+      win = require('tomtomjhj.dap').smartsplit(bufnr)
     end
 
-    api.nvim_win_call(win, function()
-        api.nvim_set_current_buf(bufnr)
-    end)
-
-    api.nvim_win_set_cursor(win, { line_num, column or 0 })
-
-    api.nvim_set_current_win(win)
+    api.nvim_win_set_cursor(0, { line_num, column or 0 })
 end
 
 return M
