diff --git a/lua/dap/session.lua b/lua/dap/session.lua
index e64e2c5..16ee261 100644
--- a/lua/dap/session.lua
+++ b/lua/dap/session.lua
@@ -505,6 +505,12 @@ local function jump_to_location(bufnr, line, column, switchbuf, filetype)
     return true
   end
 
+  function switchbuf_fn.smartsplit()
+    require('tomtomjhj.dap').smartsplit(bufnr)
+    set_cursor(0, line, column)
+    return true
+  end
+
   if switchbuf:find('usetab') then
     switchbuf_fn.useopen = switchbuf_fn.usetab
   end
