diff --git a/autoload/vimtex/fzf.vim b/autoload/vimtex/fzf.vim
index 980543b..3eeaac9 100644
--- a/autoload/vimtex/fzf.vim
+++ b/autoload/vimtex/fzf.vim
@@ -76,7 +76,7 @@ def format_number(n):
 def colorize(e):
   try:
     from colorama import Fore, Style
-    color = {'content' : Fore.WHITE,
+    color = {'content' : "",
              'include' : Fore.BLUE,
              'label' : Fore.GREEN,
              'todo' : Fore.RED}[e['type']]
@@ -87,7 +87,7 @@ def colorize(e):
       # Colour support on Windows requires Colorama
       return f"{e['title']:65}"
     else:
-      color = {'content' : "\u001b[37m",
+      color = {'content' : "",
                'include' : "\u001b[34m",
                'label' : "\u001b[32m",
                'todo' : "\u001b[31m"}[e['type']]
diff --git a/autoload/vimtex/parser/tex.vim b/autoload/vimtex/parser/tex.vim
index c3cba39..a607a37 100644
--- a/autoload/vimtex/parser/tex.vim
+++ b/autoload/vimtex/parser/tex.vim
@@ -120,8 +120,9 @@ function! vimtex#parser#tex#input_parser(line, current_file, root) abort " {{{1
         \ new_root: a:root,
         \}
 
+  let l:sanitized_line = strpart(a:line, match(a:line, g:vimtex#re#tex_input . '|^\s*\\loadglsentries'))
   " Handle \space commands
-  let l:file = substitute(a:line, '\\space\s*', ' ', 'g')
+  let l:file = substitute(l:sanitized_line, '\\space\s*', ' ', 'g')
 
   " Handle import and subfile package commands
   if l:file =~# g:vimtex#re#tex_input_import
diff --git a/autoload/vimtex/re.vim b/autoload/vimtex/re.vim
index 4e395c9..3bb726a 100644
--- a/autoload/vimtex/re.vim
+++ b/autoload/vimtex/re.vim
@@ -22,7 +22,7 @@ let g:vimtex#re#tex_input_import = '\v\\%('
 let g:vimtex#re#tex_input_package =
       \ '\v\\%(usepackage|RequirePackage)%(\s*\[[^]]*\])?\s*\{\zs[^}]*\ze\}'
 
-let g:vimtex#re#tex_input = '\v^\s*\zs%(' . join([
+let g:vimtex#re#tex_input = '\v^\s*\{?\s*\zs%(' . join([
       \   g:vimtex#re#tex_input_latex,
       \   g:vimtex#re#tex_input_import,
       \ ], '|') . ')'
